src/components/link/index.tsx

```diff
@@ -1,25 +1,22 @@
-import React, {
-  ElementType,
-  forwardRef,
-  ReactElement,
-  ReactNode,
-  useMemo,
-} from 'react';
-import { chakra, useMultiStyleConfig } from '@chakra-ui/react';
+import React, { forwardRef, ReactNode, useMemo } from 'react';
 
-interface Props {
-  children: ReactNode;
-  as?: ElementType;
-  leftIcon?: ReactElement;
-  rightIcon?: ReactElement;
-  isExternal?: boolean;
-  rel?: string;
-  target?: string;
-  variant?: 'baseLink' | 'navigationLink' | 'subNavigationLink';
-  [key: string]: unknown;
-  fontWeight?: 'regular' | 'bold';
-  ariaLabel?: string;
-  color?: string;
+import {
+  ButtonProps,
+  chakra,
+  LinkProps as ChakraLinkProps,
+  useMultiStyleConfig,
+} from '@chakra-ui/react';
+
+interface Props extends Partial<Omit<ChakraLinkProps, 'href' | 'onClick'>> {
+  leftIcon?: ReactNode;
+  rightIcon?: ReactNode;
+  ariaLabel?: string | null;
```

Comment:
One typecheck error that we will need to handle would be in this file:
https://github.com/smg-automotive/listings-web/blob/main/components/searchData/AdvancedSearchHeader.tsx#L69